### PR TITLE
firewall: T3560: Add support for MAC address groups

### PR DESCRIPTION
**Pull request by sarthurdev**
## Change Summary

Adds new group syntax `firewall group mac-group NAME mac-address xx:xx:xx:xx:xx:xx`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3560

## Component(s) name

firewall

## Proposed changes

Adds requests MAC address group support, includes op-mode update and smoketest.

## How to test

```
set firewall group mac-group TEST mac-address 00:01:02:03:04:05
set firewall name TEST rule 1 action accept
set firewall name TEST rule 1 source group mac-group TEST
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/firewall.xml.in [hunk 1]
```diff
@@ -144,6 +144,27 @@
               </leafNode>
             </children>
           </tagNode>
+          <tagNode name="mac-group">
+            <properties>
+              <help>Firewall mac-group</help>
+            </properties>
+            <children>
+              #include <include/generic-description.xml.i>
+              <leafNode name="mac-address">
```
Comment (c-po) [inline interface-definitions/firewall.xml.in:153]: Any possibility to use  the mac include file from the interfaces?

Comment (c-po) [reply interface-definitions/firewall.xml.in:153]: ` #include <include/interface/mac.xml.i>`

File: interface-definitions/include/firewall/common-rule.xml.i [hunk 2]
```diff
@@ -176,6 +176,9 @@
           <format>!&lt;MAC address&gt;</format>
           <description>Match everything except the specified MAC address</description>
         </valueHelp>
+        <constraint>
+          <validator name="mac-address-firewall"/>
```
Comment (c-po) [inline interface-definitions/include/firewall/common-rule.xml.i:180]: Why not use `<validator name="mac-address"/>`?

File: interface-definitions/include/firewall/source-destination-group-ipv6.xml.i [hunk 3]
```diff
@@ -12,6 +12,14 @@
         </completionHelp>
       </properties>
     </leafNode>
+    <leafNode name="mac-group">
```
Comment (c-po) [inline interface-definitions/include/firewall/source-destination-group-ipv6.xml.i]: This block is used twice, please create an include file

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/include/firewall/common-rule.xml.i (hunk 2 above)
Comment (sarthurdev) [reply interface-definitions/include/firewall/common-rule.xml.i:180]: We need to support ! prefix and `mac-address` is used in other modules.

File: interface-definitions/firewall.xml.in (hunk 1 above)
Comment (sarthurdev) [reply interface-definitions/firewall.xml.in:153]: Can't use interface include as we need `<multi/>`

**Pull request merged**